### PR TITLE
Feat/browser manager

### PR DESCRIPTION
**Pull request by oneofthezombies**
# PR Template

## Checklist

- [ ] E2E finished.
- [ ] Local, Self Hosted and Development environment test finished.
- [x] Local, E2E, Self Hosted, Development and Production env variable checked.
- [ ] (Backend only) DB migration file creation checked.
- [ ] **If this is a API deleted or changed**: Check side effects for API deletion or related code change.

### CODE REVIEW

**Comment by oneofthezombies** [comment]
@Ikalli @dev-kyun @yowpark please review

**Comment by oneofthezombies** [comment]
```
name: test-browser

on:
  workflow_dispatch:

jobs:
  chrome-macos-1:
    runs-on:
      browserName: chrome
      platformName: macos
      # browserVersion: latest
    record: true
    steps:
      - name: run test
        uses: dogu-actions/run-test
        with:
          checkout: true
          command: |
            python -m venv .venv
            source .venv/bin/activate && pytest web/test_web.py
        cwd: selenium/python/pytest

  chrome-macos-2:
    runs-on:
      browserName: chrome
      platformName: macos
    record: true
    steps:
      - name: run test
        uses: dogu-actions/run-test
        with:
          checkout: true
          command: |
            python -m venv .venv
            source .venv/bin/activate && pytest web/test_web.py
        cwd: selenium/python/pytest      

  chrome-macos-3:
    runs-on:
      browserName: chrome
      platformName: macos
    record: true
    steps:
      - name: run test
        uses: dogu-actions/run-test
        with:
          checkout: true
          command: |
            python -m venv .venv
            source .venv/bin/activate && pytest web/test_web.py
        cwd: selenium/python/pytest    

  # firefox-macos:
  #   runs-on:
  #     browserName: firefox
  #     platformName: macos
  #   record: true
  #   steps:
  #     - name: run test
  #       uses: dogu-actions/run-test
  #       with:
  #         checkout: true
  #         command: |
  #           cd selenium/python/pytest
  #           python -m venv .venv
  #           source .venv/bin/activate
  #           pytest web/test_web.py

  # edge-macos:
  #   runs-on:
  #     browserName: edge
  #     platformName: macos
  #   record: true
  #   steps:
  #     - name: run test
  #       uses: dogu-actions/run-test
  #       with:
  #         checkout: true
  #         command: |
  #           cd selenium/python/pytest
  #           python -m venv .venv
  #           source .venv/bin/activate
  #           pytest web/test_web.py

  # safari-macos:
  #   runs-on:
  #     browserName: safari
  #     platformName: macos
  #   record: true
  #   steps:
  #     - name: run test
  #       uses: dogu-actions/run-test
  #       with:
  #         checkout: true
  #         command: |
  #           cd selenium/python/pytest
  #           python -m venv .venv
  #           source .venv/bin/activate
  #           pytest web/test_web.py

  chrome-android:
    runs-on:
      browserName: chrome
      platformName: android
    record: true
    steps:
      - name: run test
        uses: dogu-actions/run-test
        with:
          checkout: true
          command: |
            python -m venv .venv
            source .venv/bin/activate && pytest web/test_web.py
        cwd: selenium/python/pytest
```

File: packages/typescript-private/device-server/src/browser-manager/installed-browser-finders.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+    const browserInfos = await this.findAllBrowserInstallations(deviceSerial);
+    const filteredInfos = browserInfos.filter(({ browserName: name }) => name === browserName);
+    const browserInfosWithVersion = await this.findBrowserInstallationsWithVersion(deviceSerial, filteredInfos);
+    return browserInfosWithVersion;
+  }
+
+  private async findAllBrowserInstallations(deviceSerial: Serial): Promise<{ browserPackageName: string; browserName: AndroidBrowserName }[]> {
+    const browserNameMap = new Map(Object.entries(AndroidBrowserPackageNameMap).map(([key, value]) => [value, key]));
+    const { stdout } = await Adb.shell(deviceSerial, 'pm list packages -f');
```
Comment (yowpark) [inline packages/typescript-private/device-server/src/browser-manager/installed-browser-finders.ts:49]: How about make function like Adb.getIntalledPackage() in adb.ts?

**Comment by oneofthezombies** [comment]
Reviews
[x] Delete try quit gamium app in dogu-actions/run-test
[x] Add `runs-on` type validation with readable error message

**Comment by oneofthezombies** [comment]
console_web - public

**Comment by oneofthezombies** [comment]
device-routine

**Comment by yowpark** [comment]
LGTM! Thanks for your huge work 👏

**Comment by oneofthezombies** [comment]
```
- name: case 1
	run: pwd # <doguRoutineWorkspacePath>

- name: case 2
	run: pwd # <doguRoutineWorkspacePath>/sub
	cwd: sub

- name: case 3
	uses: dogu-actions/run-test
	with:
		checkout: true # checkout to <doguRoutineWorkspacePath>
		command: pwd # <doguRoutineWorkspacePath>

- name: case 4
	uses: dogu-actions/run-test
	with:
		checkout: true # checkout to <doguRoutineWorkspacePath>
		command: pwd # <doguRoutineWorkspacePath>/sub
	cwd: sub
```

File: packages/typescript-private/device-server/src/browser-manager/installed-browser-finders.ts (hunk 1 above)
Comment (oneofthezombies) [reply packages/typescript-private/device-server/src/browser-manager/installed-browser-finders.ts]: please review https://github.com/dogu-team/dogu/pull/374/commits/cfbeb97273ce556397c44917cca02b5215bcd2a0

**Comment by oneofthezombies** [comment]
@yowpark fixed routine step working path issues https://github.com/dogu-team/dogu/pull/374/commits/0553fa419df3c5529c53c370ed84c41b418cc73b

**Comment by oneofthezombies** [comment]
@dev-kyun please review https://github.com/dogu-team/dogu/pull/374/commits/9026de710577687f5f8e27f2f601619d722ec264

**Pull request merged**